libfabric.spec

```
@@ -19,6 +19,7 @@ Group: System Environment/Libraries
 %endif
 Url: https://www.github.com/ofiwg/libfabric
 Source: https://github.com/ofiwg/%{name}/archive/v%{dl_version}.tar.gz
+Patch0: DAOS-12407-ofi-patch.diff
```

Instead of manually uploading the patch you could simply pick it up from there:
```suggestion
Patch0: https://github.com/ofiwg/libfabric/commit/d438a9bcdce31c9950e009f4a737b3809ba0d445.patch
```

---

The preferred practice is to include a copy of the patch in the repo.  Providing a URL to a patch (such you suggest) is also great for documenting it's origination, but we don't want to be depending on the original location always being present.

A similar argument could be made for the *Source*:(s) also, but they are bigger binary patches which we have experience with keeping in git and the experience is not good.  An upstream missing source is also a fundamental problem that pretty much voids the entire packaging effort, vs. just missing a needed patch.

---

I mean unless they rewrite the entire repo history I guess that commit could disappear but it seems unlikely? :) or we'd probably have bigger problems before that.

OK yes I get the argument for keeping a copy of the sources however but yeah since they all come from the same github location it would seem pretty much the same to me.

---

Or upstream simply decides that GitHub is not the place for them and moves the project and closes their GH account.